04/implement-bipartition-graph.checkpoint.md

```diff
@@ -0,0 +1,348 @@
+# Implement Possible Graph Bipartition
+
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: 5875e4c4-dab1-427e-935b-c5983b45bd92
+* title: Possible Graph Bipartition
+* points: 3
+### !question
+For this exercise, create a function `possible_bipartition` which takes in an adjacency list representing a graph of puppies, `dislikes`, to determine if the puppies can be divided into two groups where no two puppies that are known for fighting each other are in the same group.  
+
+Given a set of N puppies, we would like to split them into two groups of any size to use two play areas.
+
+Formally, if `dislikes[i] = [a, b]`, it means puppy `i` is not allowed to put in the same group as puppy `a` or puppy `b`.
+
+Dislike is mutual. If puppy `a` dislikes puppy `b`, puppy `b` also dislikes puppy `a`.
+
+Return `True` if and only if it is possible to split the puppies into two groups where no fighting will occur. Otherwise, return `False`.
+
+**Example 1**
+```
+Input:
+dislikes = { 
+    "Fido": [],
+    "Nala": ["Cooper", "Spot"],
+    "Cooper": ["Nala", "Bruno"],
+    "Spot": ["Nala"],
+    "Bruno": ["Cooper"]
+}
+
+Output: True
+
+Explanation:
+Fido can be placed in Group 1.
+Nala can be placed in Group 1.
+Cooper can be placed in Group 2.
+Spot can be placed in Group 2.
+Bruno can be placed in Group 1.
+
+None of the pups who would fight with each other are placed in the same group.
```

Comment:
Same here; just ensure any changes from `bipartition-graph.md` are copied over to this document.